Libraries/LibWeb/HTML/Scripting/Fetching.cpp

```diff
@@ -85,8 +85,8 @@ ByteString module_type_from_module_request(JS::ModuleRequest const& module_reque
 WebIDL::ExceptionOr<URL::URL> resolve_module_specifier(Optional<Script&> referring_script, ByteString const& specifier)
 {
     // 1. Let settingsObject and baseURL be null.
-    Optional<EnvironmentSettingsObject&> settings_object;
-    Optional<URL::URL const&> base_url;
+    GC::Ptr<EnvironmentSettingsObject> settings_object;
+    Optional<URL::URL> base_url;
```

Comment:
This makes me wonder why we use `Optional<T&>` at all

Reply:
:thinking: Perhaps we should try to make it harder to assign an xvalue or rvalue to an optional T&? @alimpfard any thoughts on how to check for that in AK?

Reply:
```c++
    template<typename U = T>
    requires(CanBePlacedInOptional<U&>)
    ALWAYS_INLINE Optional(U const&& value) = delete;
```

?

Reply:
const rvalue is a nonsense qualifier, but otherwise yeah, that looks like it would do the trick. + some test coverage, that would make a nice PR